test/known_problems/should_pass/nested_pattern_match.erl

```diff
@@ -0,0 +1,12 @@
+-module(nested_pattern_match).
+-compile([debug_info]).
+-export([f/1]).
+
+-record(s, {a :: c | d}).
+-record(r, {a :: #s{} | boolean()}).
+
+-spec f(#r{}) -> boolean().
+f(R) -> if
+    (R#r.a)#s.a == c -> true;
```

Comment:
I think this is a guards specific problem. Exceptions (e.g. runtime type errors) in guards are treated as just no match and this is used in practice in code such as this. So I think we should complain only if it can never match, as you say.

Nested patterns for pattern matching works, I hope, doesn't it?